src/tlo/methods/rti.py

```diff
@@ -2500,17 +2487,104 @@ def rti_assign_injuries(self, number):
         # Finally return the injury description information
         return inj_df
 
-    def do_rti_diagnosis_and_treatment(self, person_id):
-        """Things to do upon a person presenting at a Non-Emergency Generic HSI if they have an injury."""
-        df = self.sim.population.props
-        persons_injuries = df.loc[person_id, RTI.INJURY_COLUMNS]
-        if pd.isnull(df.at[person_id, 'cause_of_death']) and not df.at[person_id, 'rt_diagnosed']:
-            if len(set(RTI.INJURIES_REQ_IMAGING).intersection(persons_injuries)) > 0:
-                self.rti_ask_for_imaging(person_id)
-            df.at[person_id, 'rt_diagnosed'] = True
-            self.rti_do_when_diagnosed(person_id=person_id)
-            if df.at[person_id, 'rt_in_shock']:
-                self.rti_ask_for_shock_treatment(person_id)
+    def _common_first_appt_steps(
+        self,
+        patient_id: int,
+        patient_details: NamedTuple = None,
+    ) -> IndividualPropertyUpdates:
+        """
+        Shared logic steps that are used by the RTI module when a generic HSI
+        event is to be scheduled.
+        """
+        df_updates = {}
+        # Things to do upon a person presenting at a Non-Emergency Generic
+        # HSI if they have an injury.
+        persons_injuries = [
+            getattr(patient_details, injury) for injury in RTI.INJURY_COLUMNS
+        ]
+        if (
+            pd.isnull(patient_details.cause_of_death)
+            and not patient_details.rt_diagnosed
+        ):
+            if set(RTI.INJURIES_REQ_IMAGING).intersection(set(persons_injuries)):
```

Comment:
Oh interesting! No need to change then, I vaguely thought there was some sort of recommendation in PEP8 to avoid implicit falseyness tests like this but I checked and there isn't so I think it's just me imagining it and better to stick with the linter recommendations